lib/Zonemaster/CLI.pm

```diff
@@ -517,13 +518,24 @@ sub add_fake_delegation {
         if ($ip) {
             push @{ $data{ $self->to_idn( $name ) } }, $ip;
         }
+        elsif (Zonemaster::Zone->new( { name => $domain } )->is_in_zone( $name )) {
+            say STDERR "--ns in zone nameservers must be a name/ip pair.";
+            exit( 1 );
+        }
         else {
             my $n = $self->to_idn( $name );
             my @ips = Net::LDNS->new->name2addr($n);
-            push @{ $data{$n} }, $_ for @ips;
+            if ( @ips ) {
+                push @{ $data{$n} }, $_ for @ips;
+            }
+            else {
+                say STDERR "--ns ${name} nameserver has no IP address.";
+                exit( 1 );
+            }
```

Comment:
I do not think this is the correct design. The error should be like any other error. It should be caught be a test case and reported. With the solution above, Zonemaster will not check the other nameservers provided.